doc/services/zbus/index.rst

```diff
@@ -0,0 +1,356 @@
+No user's thread must have a priority higher than the zbus event dispatcher.
+Synchronous observers must have non-blocking and quick-as-possible implementations.
+In asynchronous observer scenarios, producers must have a lower priority than consumers.
+
+Message delivery sequence
+-------------------------
+
+The listeners (synchronous observers) will follow the channel definition sequence as the notification and message consumption sequence. However, the subscribers, as they have an asynchronous nature, all will receive the notification as the channel definition sequence but only will consume the data when they assume the CPU again, so the delivery respects the order, but the reaction of it will be oriented by the priority assigned to the subscribers.
+
+Implementation
+**************
+
+Zbus operation depends on channels and observers. Therefore, it is necessary to determine its message and observers list during the channel definition. A message is a regular C struct; the observer can be a subscriber (asynchronous) or a listener (synchronous). Channels can have a combination of the following characteristics:
+
+* ``Validator function`` makes a channel accepts only valid messages;
+* ``Real-only flag`` makes a channel read-only.
+
+.. warning::
+
+    The ``Validator function`` will not take effect in read-only channels.
+
+The following code defines and initializes a read-only channel and its dependencies. This channel represents a firmware version, for example.
+
+.. code-block:: c
+
+    struct version {
+            uint8_t major;
+            uint8_t minor;
+            uint16_t build;
+    };
+
+    ZBUS_CHAN_DEFINE(version,	       /* Name */
+             true,		       /* Read only */
+             struct version,       /* Message type */
+
+             NULL,		       /* Validator */
+             ZBUS_OBSERVERS_EMPTY, /* observers */
+             ZBUS_MSG_INIT(.major = 0, .minor = 1,
+                       .build = 2) /* Initial value major 0, minor 1, build 1023 */
+    );
+
+
+The following code defines and initializes a regular channel and its dependencies. This channel exchanges accelerometer data, for example.
+
+.. code-block:: c
+
+    struct acc_msg {
+            int x;
+            int y;
+            int z;
+    };
+
+    ZBUS_CHAN_DEFINE(acc_data,				     /* Name */
+             false,					     /* Read only */
+             struct acc_msg,			     /* Message type */
+
+             NULL,					     /* Validator */
+             ZBUS_OBSERVERS(my_listener, my_subscriber), /* observers */
+             ZBUS_MSG_INIT(.x = 0, .y = 0, .z = 0)	     /* Initial value {0} */
+    );
+
+    void listener_callback_example(struct zbus_channel *chan)
+    {
+            struct acc_msg acc = {0};
+            if (&acc_data == chan) {
+                    zbus_chan_read(chan, &acc, K_NO_WAIT);
+                    LOG_DBG("From listener -> Acc x=%d, y=%d, z=%d", acc.x, acc.y, acc.z);
+            }
+    }
```

Comment:
Should there be a `ZBUS_LISTENER_DEFINE(my_listener, listener_callback_example);` here?